Framework/Kernel/src/FacilityInfo.cpp

```diff
@@ -59,6 +60,15 @@ void FacilityInfo::fillZeroPadding(const Poco::XML::Element *elem) {
   }
 }
 
+/// Called from constructor to fill the allowNoPrefix flag
+void FacilityInfo::fillAllowNoPrefix(const Poco::XML::Element *elem) {
+  std::string allowNoPrefixStr = elem->getAttribute("allowNoPrefix");
+  if (allowNoPrefixStr == "True")
+    m_allowNoPrefix = true;
+  else
+    m_allowNoPrefix = false;
```

Comment:
Instead of the branch, what do you think about just storing the result of the comparison?